### PR TITLE
Add glossary index

### PR DESCRIPTION
**Pull request by mingoscd**

### CODE REVIEW

**Comment by BirgitBader** [comment]
Hi @mingoscd The CI checks on your PR have failed, see attached screen. Please recheck.
bildschirmfoto 2015-07-10 um 07 58 42

**Comment by mingoscd** [comment]
The problem was that the links with the missing letters didn't exists.
The last commit solves that but the glossary only shows the existing letters

**Comment by otrosien** [comment]
@mingoscd general git advice - avoid committing into dev, but create feature branches and send pull requests from there. See http://nvie.com/posts/a-successful-git-branching-model/

**Comment by mingoscd** [comment]
@otrosien thanks for the advise :) . That is what I'm trying to do. This branch is not develop, is domingo/develop but the next time I will name it with the feature name to avoid confusions.

File: _sass/bootstrap/_button-groups.scss [hunk 1]
```diff
@@ -27,7 +27,7 @@
   .btn + .btn-group,
   .btn-group + .btn,
   .btn-group + .btn-group {
-    margin-left: -1px;
```
Comment (choffmeister) [inline _sass/bootstrap/_button-groups.scss]: Please do not alter vendor files (this makes upgrading external dependencies harder). If you have to override such stuff, create a custom CSS class in our files and use this class instead.

File: _layouts/page.html [hunk 2]
```diff
@@ -68,6 +68,13 @@
             </div>
           {% endunless %}
         {% endif %}
+        {% if page.title == 'Glossary' %}
+          <div class="btn-group" role="group">
+          {% for letter in page.categories %}
+            <button type="button" class="btn btn-default"><a href="#{{ letter }}">{{ letter | upcase }}</a></button>
+          {% endfor %}
+          </div>
+        {% endif %}
```
Comment (choffmeister) [inline _layouts/page.html]: Being the glossary page is not a general purpose and hence does belong here in the general page layout file.

I guess the problem you want to solve, is that the glossary button shall appear above the headline? If so, you can do it like this:

```html
---
layout: page
title: My title
title_hidden: true
---

<!-- What needs to appear before the page title -->
<div class="btn-group" role="group">
  {% for letter in page.categories %}
    <button type="button" class="btn btn-default"><a href="#{{ letter }}">{{ letter | upcase }}</a></button>
  {% endfor %}
</div>

<!-- Insert the page title markup manually -->
<div class="page-header">
  <h1>{{ page.title }}</h1>
</div>

Foobar...
```

Comment (choffmeister) [inline _layouts/page.html]: Or, if you just want to reuse this glossary button stuff, create another layout called `glossary` that uses the `page` layout and adds the buttons.

**Comment by choffmeister** [comment]
@mingoscd Now I am almost fine with it. But could you please squash your commits. You have now five commits for adding 7 lines into one file.

**Comment by BirgitBader** [comment]
@choffmeister Is this okay now from your perspective? If yes, please merge.

**Pull request merged**